magento/magento2
# Fall back to parent directory if composer.json not found in module directory

### Description
This change allows modules to follow the [pds/skeleton](https://github.com/php-pds/skeleton) standard and have their source code in a `src` subdirectory of the repository.

### Manual testing scenarios

1. create a module with `composer.json` and `registration.php` in the root directory and source code in `src`

    registration.php
    ```
    \Magento\Framework\Component\ComponentRegistrar::register(
        \Magento\Framework\Component\ComponentRegistrar::MODULE,
        'Vendor_Module',
        __DIR__ . '/src'
    );
    ```

    composer.json
    ```
      "autoload": {
        "files": [
          "registration.php"
        ],
        "psr-4": {
          "Vendor\\Module\\": "src"
        }
      },
      "suggest": {
        "vendor/module-sample-data": "Sample Data version:1.0.0"
      },
    ```
2. call `bin/magento sampledata:deploy` and see if `vendor/module-sample-data` is going to be installed.

### Contribution checklist
